platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -402,15 +445,18 @@ protected BundleContext getContext() {
   /** {@inheritDoc}. */
   @Override
   public List<Map<String, Object>> getPluginsForApplication(String appName) {
-    List<Map<String, Object>> returnValues = new ArrayList<>();
-
-    for (ApplicationPlugin plugin : applicationPlugins) {
-      if (plugin.matchesAssocationName(appName)) {
-        returnValues.add(plugin.toJSON());
+    try {
+      List<Map<String, Object>> returnValues = new ArrayList<>();
+      for (ApplicationPlugin plugin : applicationPlugins) {
+        if (plugin.matchesAssocationName(appName)) {
+          returnValues.add(plugin.toJSON());
+        }
       }
+      return returnValues;
+    } catch (Exception e) {
+      LOGGER.error("Could not get plugins for application", e);
+      throw new ApplicationServiceBeanException("Could not get plugins for application");
```

Comment:
Should we include the exception in this new `ApplicationServiceBeanException`?

Reply:
Nope. That will send a stacktrace to the UI. We should redact that information. The full details will be reviewable by those with access to the backend via the above error log.